flybywiresim/a32nx
# feat(engine): Cold Start - Primary Parameters

Fixes #5001
Fixes #4621 
Fixes #2351

## Summary of Changes

This PR features a realistic cold engine start-up process for the primary parameters (N1, N2, EGT and FF). Engine spools as a function of pressure and temperature until Idle condition is reached.

No additional visuals have been implemented/ changed, except for the small delay in the start-valve opening (SD Engine page).

Sound XML file updated and adapted to new LVars (courtesy of boris#8008).

New Engine LVars included in the FDR code (courtesy of AGuther [Z+2]#0001).

## Screenshots (if necessary)


Before engine start, all parameters are stable at 0 except EGT which is set at ambient temperature.

![EWD-1](https://user-images.githubusercontent.com/70781153/121480736-ac3aa280-c9cb-11eb-8442-6f8e1a735918.PNG)

Expected behavior of N2, EGT and FF as a function of time during start-up:

![image](https://user-images.githubusercontent.com/70781153/121481589-924d8f80-c9cc-11eb-8f2e-f801f57af93f.png)


## References

- Several YT videos of real Leap start-ups
- Discord user FlyingCarpet7#7502




## Additional context



Discord username for Engine Model (if different from GitHub): TazX [Z+2]#0405
Discord username for Sound Model (if different from GitHub): boris#8008

## Testing instructions

For testing, check that the start-up procedure is as expected in the figures above for different set of environmental conditions:

- High and low altitude airports
- High and low ambient temperatures

Other than the process itself:
- check that EGT initial temperature starts at ambient temperature
- check that the transition from start-up to Idle is smooth (since two separate functions are being used)


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
